src/components/Auth.jsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import axios from 'axios';
+
+export default class Auth extends React.Component {
+    componentDidMount() {
+        axios.get('/auth')
+        .then((response) => {
+            console.log(response);
+            if (response.status === 200) {
+                if (window.location.pathname !== '/dashboard') {
+                    window.location = '/dashboard'
+                } 
+            }
+        }).catch((err) => {
+            console.log(err);
+            
+            if (window.location.pathname !== '/') {
+                window.location = '/';
+            }
+        })
+    }
+    render() {
+        return null;
```

Comment:
I would move this into an action creator rather than a renderless component. Doesn't really make sense to have this as a component. You can have actions that are called as soon as the app starts. They don't have to be called from a component.